src/backend/tests/tasks.test.ts

```diff
@@ -202,4 +202,55 @@ describe('Tasks', () => {
       );
     });
   });
+
+  describe('deleteTask', () => {
+    const mockTaskId = '4';
+
+    test('delete task fails when task does not exist', async () => {
+      jest.spyOn(prisma.task, 'findUnique').mockResolvedValue(null);
+
+      await expect(() => TasksService.deleteTask(batman, mockTaskId)).rejects.toThrow(
+        new NotFoundException('Task', mockTaskId)
+      );
+
+      expect(prisma.task.findUnique).toHaveBeenCalledTimes(1);
+    });
+
+    test('delete task fails when wbs element does not exist', async () => {
+      jest.spyOn(prisma.task, 'findUnique').mockResolvedValue(taskSaveTheDayDeletedPrisma);
+      jest.spyOn(prisma.wBS_Element, 'findUnique').mockResolvedValue(null);
+
+      await expect(() => TasksService.deleteTask(batman, mockTaskId)).rejects.toThrow(
+        new NotFoundException('WBS Element', taskSaveTheDayDeletedPrisma.wbsElementId)
+      );
+
+      expect(prisma.task.findUnique).toHaveBeenCalledTimes(1);
+      expect(prisma.wBS_Element.findUnique).toHaveBeenCalledTimes(1);
+    });
+
+    test('delete task fails when user does not have permission', async () => {
+      jest.spyOn(prisma.task, 'findUnique').mockResolvedValue(taskSaveTheDayDeletedPrisma);
+      jest.spyOn(prisma.wBS_Element, 'findUnique').mockResolvedValue(prismaWbsElement1);
+
+      await expect(() => TasksService.deleteTask(wonderwoman, mockTaskId)).rejects.toThrow(
+        new AccessDeniedException('You do not have permission to delete this task')
```

Comment:
if you want this to be the message you have to alter the message given in the actual service function